css-mixins-1/Overview.bs

```diff
@@ -575,3 +597,90 @@ Issue: Should we indeed use ''@nest'' for this purpose?
 	since only custom properties
 	and the '@function/result' descriptor
 	are relevant.
+
+<!-- Big Text: appendix
+
+ ███▌  ████▌  ████▌  █████▌ █    █▌ ████▌  ████ █     █
+▐█ ▐█  █▌  █▌ █▌  █▌ █▌     █▌   █▌ █▌  █▌  ▐▌   █   █ 
+█▌  █▌ █▌  █▌ █▌  █▌ █▌     ██▌  █▌ █▌  █▌  ▐▌    █ █  
+█▌  █▌ ████▌  ████▌  ████   █▌▐█ █▌ █▌  █▌  ▐▌     █   
+█████▌ █▌     █▌     █▌     █▌  ██▌ █▌  █▌  ▐▌    █ █  
+█▌  █▌ █▌     █▌     █▌     █▌   █▌ █▌  █▌  ▐▌   █   █ 
+█▌  █▌ █▌     █▌     █████▌ █▌   ▐▌ ████▌  ████ █     █
+-->
+
+Appendix: The <<syntax>> Production {#syntax}
+=============================================
+
+The <<syntax>> production represents a [=syntax definition=],
+which may be used to impose a type
+on [=function parameters=],
+[=function dependencies=],
+or [=custom function=] return values.
+
+<pre class="prod def" nohighlight>
+
+<dfn><<type-name>></dfn> = angle | color | custom-ident | image | integer |
+	length | length-percentage | number |
+	percentage | resolution | string | time |
+	url | transform-function
+<dfn><<combinator>></dfn> = '|'
+<dfn><<multiplier>></dfn> = [ '#' | '+' ]
+<dfn><<syntax-component-name>></dfn> = '<' <<type-name>> '>' | <<custom-ident>>
+<dfn><<syntax-component>></dfn> = <<syntax-component-name>> <<multiplier>>? |
+	'<' transform-list '>'
+<dfn><<syntax>></dfn> = '*' | <<syntax-component>> [ <<combinator>> <<syntax-component>> ]+
+
+</pre>
+
+A <<syntax-component>> consists of either a <<type-name>>,
+which maps to one of the [=supported syntax component names=];
+or a <<custom-ident>>, which represents any [=keyword=].
+Additionally,
+a <<syntax-component>> may contain a [[css-properties-values-api-1#multipliers|multiplier]],
+which indicates a [=list=] of values.
+
+Note: This means that <code>&lt;length&gt;</code>
+	and <code>length</code> are two different types:
+	the former describes a <<length>>,
+	whereas the latter describes a [=keyword=] <code>length</code>.
+
+Multiple <<syntax-component>>s may be [[css-properties-values-api-1#combinator|combined]]
+with a <code>|</code> <<delim-token>>,
+causing the syntax components to be matched
+against a value
+in the specified order.
+
+<div class='example'>
+	<pre class='lang-css'>
+	&lt;percentage&gt; | &lt;number&gt; | auto
+	</pre>
+	The above, when parsed as a <<syntax>>,
+	would accept <<percentage>> values,
+	<<number>> values,
+	as well as the keyword <code>auto</code>.
+</div>
+
+<div class='example'>
+	<pre class='lang-css'>
+	red | &lt;color&gt;
```

Comment:
Is the order important for this example? I don't think it should be, but not sure if we need to clarify that.

Reply:
The order matters when the thing using the syntax definition (and the values produced using that syntax definition) wants to act differently on identifiers vs colors. For example, a registered custom property with `"red | <color>"` would compute `red` to `red`, but `blue` to `rgb(0, 0, 255)`.

For custom functions I don't think it actually matters, but I wrote this section with the intent to eventually move it somewhere more general.